deltachat-ios/Handler/AppStateRestorer.swift

```diff
@@ -33,9 +38,34 @@ class TabBarRestorer: NSObject, UITabBarControllerDelegate {
 
     func tabBarController(_ tabBarController: UITabBarController, didSelect viewController: UIViewController) {
         let activeTab = tabBarController.selectedIndex + offsetKey
+
+        if let tab = Tab(rawValue: activeTab), tab != .chatTab {
+            resetLastActiveChat()
+        }
+
         UserDefaults.standard.set(activeTab, forKey: lastActiveTabKey)
         UserDefaults.standard.synchronize()
     }
 
+    private func storeChat(chatId: Int?) {
+        let value = chatId ?? -1
+        UserDefaults.standard.set(value, forKey: lastActiveChatId)
+        UserDefaults.standard.synchronize()
```

Comment:
From the docs: 
 
-synchronize is deprecated and will be marked with the API_DEPRECATED macro in a future release.
-synchronize blocks the calling thread until all in-progress set operations have completed. This is no longer necessary. Replacements for previous uses of -synchronize depend on what the intent of calling synchronize was. If you synchronized...
     - ...before reading in order to fetch updated values: remove the synchronize call
     - ...after writing in order to notify another program to read: the other program can use KVO to observe the default without needing to notify
     - ...before exiting in a non-app (command line tool, agent, or daemon) process: call CFPreferencesAppSynchronize(kCFPreferencesCurrentApplication)
     - ...for any other reason: remove the synchronize call


I tested the PR without this method call and it works fine on iOS 13. I didn't test with older devices though

Reply:
Oh wow. Didn't know. That's actually good news :) 

I will delete the .synchronize thing,